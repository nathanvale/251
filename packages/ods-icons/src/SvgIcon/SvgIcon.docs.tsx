/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Text, TextLink } from "@origin-digital/ods-core";
import { SvgIcon, SvgIconProps } from "./SvgIcon";

export const docs: ComponentDocs<SvgIconProps> = {
  category: "Atomic",
  componentName: "SvgIcon",
  description: (
    <Text>
      If you need a custom SVG icon, not available in the{" "}
      <TextLink href="#/foundations/icons">ODS Icon library</TextLink>, you can
      use the <code>&lt;SvgIcon/&gt;</code> wrapper. This component extends the
      native <code>&lt;svg&gt;</code> element.
    </Text>
  ),
  propDescriptions: {
    children: "Svg nodes passed into the SVG element",
    color:
      "The color of the component. It supports theme colors that make sense for this component.",
    tone: `Deprecated, use "color" prop instead.`,
    size:
      "The fontSize applied to the icon, xsmall (20px) | small (24px) | medium (48px) | large (64px). It can also be configured to inherit font size.",
    viewBox:
      "Allows you to redefine what the coordinates units mean inside an SVG element",
    titleAccess:
      "Provides a human-readable title for the element that contains it.",
  },
  migrationGuide: false,
  examples: {
    default: {},
    additional: [
      {
        label: "Creating a custom svg",
        description: (
          <Text>
            To create your custom svg simply pass in the svg{" "}
            <code>&lt;path&gt;</code> as children. (Mood icon{" "}
            <code>&lt;path&gt;</code> from the free{" "}
            <TextLink href="https://material.io/resources/icons/?search=mood&icon=mood&style=baseline">
              Material icons
            </TextLink>{" "}
            library.)
          </Text>
        ),
        Code: () => (
          <SvgIcon size="large">
            <path d="M0 0h24v24H0z" fill="none" />
            <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z" />
          </SvgIcon>
        ),
      },
      {
        label: "Creating a custom svg with a redefined view box",
        description: (
          <Text>
            To redefine what the svg coordinates units mean pass in a custom{" "}
            <code>viewBox</code>. (Heart icon <code>&lt;path&gt;</code> with a{" "}
            <code>viewBox="0 0 48 48"</code> from the free{" "}
            <TextLink href="https://icomoon.io/">icomoon</TextLink> library.)
          </Text>
        ),
        Code: () => (
          <SvgIcon color="critical" size="large" viewBox="0 0 48 48">
            <path d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z" />
          </SvgIcon>
        ),
      },
    ],
  },
  snippets: [],
};
