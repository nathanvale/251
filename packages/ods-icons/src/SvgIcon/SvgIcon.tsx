import React from "react";
import {
  SvgIcon as MUISvgIcon,
  SvgIconProps as MuiSvgIconProps,
} from "@material-ui/core";
import {
  OptionalTrackableProps,
  SvgIconToneVariants,
  SvgIconSizeVariants,
} from "@origin-digital/ods-types";
import { useSvgIconStyles } from "../_private/useSvgIconStyles";

export interface SvgIconProps extends OptionalTrackableProps {
  children?: React.ReactNode;
  tone?: SvgIconToneVariants;
  size?: SvgIconSizeVariants;
  muiProps?: MuiSvgIconProps;
  titleAccess?: string;
  viewBox?: string;
}

export const defaultColor = "secondary";
export const defaultsize = "small";
export const defaultViewBox = "0 0 24 24";

export const SvgIcon = React.forwardRef<any, SvgIconProps>((props, ref) => {
  const {
    "data-id": dataId,
    children,
    tone = defaultColor,
    size = defaultsize,
    muiProps,
    ...rest
  } = props;
  const svgIconStyles = useSvgIconStyles({ tone, size });
  return (
    <MUISvgIcon
      data-id={dataId}
      className={svgIconStyles}
      ref={ref}
      {...rest}
      {...muiProps}
    >
      {children}
    </MUISvgIcon>
  );
});

SvgIcon.defaultProps = {
  "data-id": "svg-icon",
  tone: defaultColor,
  size: defaultsize,
  viewBox: defaultViewBox,
};

SvgIcon.displayName = "SvgIcon";
