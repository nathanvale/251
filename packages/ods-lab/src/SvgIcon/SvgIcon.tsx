import * as React from "react";
import MuiSvgIcon, {SvgIconProps as ISvgIcon} from "@material-ui/core/SvgIcon";

export interface SvgIconProps extends ISvgIcon {}

export const SvgIcon = (props: ISvgIcon) => (
  <MuiSvgIcon {...props}>{props.children}</MuiSvgIcon>
);
