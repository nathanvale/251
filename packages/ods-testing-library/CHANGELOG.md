# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
