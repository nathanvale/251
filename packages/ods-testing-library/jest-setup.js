import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
