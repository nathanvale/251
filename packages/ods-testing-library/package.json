{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
