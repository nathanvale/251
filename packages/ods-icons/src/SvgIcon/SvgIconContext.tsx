import { createContext } from "react";
import { SvgIconToneVariants } from "@origin-digital/ods-types";
import { defaultColor } from "./SvgIcon";

export const SvgIconContext = createContext<SvgIconToneVariants>(defaultColor);
