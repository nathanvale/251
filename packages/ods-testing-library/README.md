# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
