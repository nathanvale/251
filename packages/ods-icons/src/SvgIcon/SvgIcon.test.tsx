import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { SvgIcon } from "./SvgIcon";

test("it renders a svg", () => {
  const { container } = render(
    <SvgIcon>
      <path d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z" />
    </SvgIcon>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders a large svg", () => {
  const { container } = render(
    <SvgIcon size="large">
      <path d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z" />
    </SvgIcon>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders a svg with color", () => {
  const { container } = render(
    <SvgIcon color="critical">
      <path d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z" />
    </SvgIcon>
  );
  expect(container.firstChild).toMatchSnapshot();
});
