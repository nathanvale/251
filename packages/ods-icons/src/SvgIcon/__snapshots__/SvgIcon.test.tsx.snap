// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a large svg 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root svg-icon-secondary-large-svg-icon-2"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z"
  />
</svg>
`;

exports[`it renders a svg 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-1"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z"
  />
</svg>
`;

exports[`it renders a svg with color 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root svg-icon-critical-small-svg-icon-3"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M35.399 3c-5.045 0-9.387 4.104-11.398 8.391-2.013-4.286-6.355-8.391-11.401-8.391-6.955 0-12.6 5.646-12.6 12.601 0 14.149 14.273 17.859 24.001 31.848 9.195-13.902 23.999-18.15 23.999-31.848 0-6.956-5.645-12.601-12.601-12.601z"
  />
</svg>
`;
